test/web-platform-tests/start-wpt-server.js

```diff
@@ -30,39 +30,45 @@ module.exports = ({ toUpstream = false } = {}) => {
   const configPath = configPaths[configType];
   const config = configs[configType];
 
-  return dnsLookup("web-platform.test").then(
-    () => {
-      const configArg = path.relative(path.resolve(wptDir), configPath);
-      const args = ["./wpt.py", "serve", "--config", configArg];
-      const subprocess = childProcess.spawn("python", args, {
-        cwd: wptDir,
-        stdio: "inherit"
-      });
+  const configArg = path.relative(path.resolve(wptDir), configPath);
+  const args = ["./wpt.py", "serve", "--config", configArg];
+  let subprocess;
+
+  const result = new Promise((resolve, reject) => {
+    subprocess = childProcess.spawn("python", args, {
+      cwd: wptDir,
+      stdio: "inherit"
+    });
+
+    subprocess.on("error", e => {
+      reject(new Error(`Error starting python server process: ${e.message}`));
+    });
 
-      return new Promise((resolve, reject) => {
-        subprocess.on("error", e => {
-          reject(new Error("Error starting python server process:", e.message));
-        });
+    process.on("exit", () => {
+      // Python doesn't register a default handler for SIGTERM and it doesn't run __exit__() methods of context
+      // managers when it gets that signal. Using SIGINT avoids this problem.
+      subprocess.kill("SIGINT");
+    });
 
-        resolve(Promise.all([
+    dnsLookup("web-platform.test")
```

Comment:
Why did this get rearranged to happen after server startup? I prefer surfacing these misconfiguration errors as soon as possible.

Reply:
This is so that `subprocess` can be accessed synchronously as the `child` property of the returned `Promise` (same as `util.promisify(childProcess.exec`).

Reply:
Hmm, that doesn't seem like a worthwhile tradeoff to me. It makes the test running experience worse just to make our code a bit simpler.

Reply:
Well, if I don’t do this, then if the tests finish before DNS lookup completes, then the server will still start, but won’t be terminated in the `after` hook, because `serverProcess` will be `undefined` in the `after` hook: <https://github.com/jsdom/jsdom/pull/2905#discussion_r394082401>.